.github/workflows/ci.yaml

```diff
@@ -7,64 +7,120 @@ on:
+
+      # Downloaded artifacts lose their prior permission settings
+      - name: Adjust parser permissions
+        run: chmod +x parser/target/x86_64-unknown-linux-musl/release/parser
+
       - name: Set up Cloud SDK
+        if: matrix.connector == 'source-gcs'
         uses: google-github-actions/setup-gcloud@master
         with:
           project_id: ${{ secrets.GCP_PROJECT_ID }}
           service_account_key: ${{ secrets.GCP_SERVICE_ACCOUNT_KEY }}
           export_default_credentials: true
+
       - name: Configure AWS credentials from Test account
+        if: matrix.connector == 'source-kinesis' || matrix.connector == 'source-s3'
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: us-east-2
-      - name: Connector integration tests
-        run: make integration-tests
+
+      - name: Login to GitHub package docker registry
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | \
+            docker login --username ${{ github.actor }} --password-stdin ghcr.io
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build ${{ matrix.connector }} Docker Image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ${{ matrix.connector }}/Dockerfile
+          load: true
+          tags: ghcr.io/estuary/${{ matrix.connector }}:test
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: ${{ matrix.connector }} integration tests
+        if: matrix.connector != 'source-hello-world'
         env:
           GCP_SERVICE_ACCOUNT_KEY: ${{ secrets.GCP_SERVICE_ACCOUNT_KEY }}
           AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           DEFAULT_AWS_REGION: ${{ secrets.DEFAULT_AWS_REGION }}
           AWS_DEFAULT_OUTPUT: json
+        run: CONNECTOR=${{ matrix.connector }} VERSION=test ./tests/run.sh;
 
-      - name: Login to GitHub package docker registry
-        run: |
-          echo "${{ secrets.GITHUB_TOKEN }}" | \
-            docker login --username ${{ github.actor }} --password-stdin ghcr.io
-      - name: Tidy up, and ensure that generated files are unchanged.
-      # Tidy up before further invocations using `git describe --dirty`
-        run: |
-          # In case any tests left their data laying around
-          git clean --force -d
-          git diff
-          [[ -z "$(git status --porcelain)" ]] || exit 1
-        shell: bash
-      - run: make push-all
-
+      - name: Push ${{ matrix.connector }} image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ${{ matrix.connector }}/Dockerfile
+          push: true
+          tags: ghcr.io/estuary/${{ matrix.connector }}:${{ steps.prep.outputs.tag }}
```

Comment:
We tag any successful connector build with the git sha and push it. This *does not* push a `latest`/`dev`/whatever tag on `main` builds, but we were not doing that before either. 

Whenever we decide that we want to do that, I think we'd want to make that a distinct separate job that depends on `build_connectors`.